projects/pastanaga/src/lib/controls/filtered-checkbox-group/filtered-checkbox-group.component.ts

```diff
@@ -0,0 +1,159 @@
+    _filterPlaceholder = 'pastanaga.filter';
+
+    @Output() selection: EventEmitter<string[]> = new EventEmitter();
+    @Output() checkboxesChange: EventEmitter<ControlModel[]> = new EventEmitter();
+
+    _checkboxes: ControlModel[] = [];
+    isFiltered = false;
+    isAllSelected = false;
+
+    filtered: ControlModel[] = [];
+    selectedLetter = '';
+    totalCount = 0;
+    totalSelected = 0;
+    totalFiltered = 0;
+    filter = '';
+    selectAllLabel = '';
+    keyUp: Subject<string> = new Subject();
+    terminator = new Subject();
+    @ViewChild(CdkVirtualScrollViewport, { static: true }) scrollViewport?: CdkVirtualScrollViewport;
+    trackById = (a, b) => (a.id === b.id);
+
+    constructor(
+        private translate: TranslatePipe,
+        private cdr: ChangeDetectorRef,
+    ) {}
+
+    ngOnInit() {
+        this.id = !this.id ? `fieldset-filtered-checkbox-group-${nextId++}` : `${this.id}-filtered-checkbox-group`;
+        this.keyUp.pipe(
+            takeUntil(this.terminator),
+            debounceTime(500),
+            distinctUntilChanged()
+        ).subscribe(term => {
+            this.filter = term;
+            this.filterByTerm();
+            if (!!this.scrollViewport) {
+                this.scrollViewport.scrollToIndex(0);
+            }
+        });
+    }
+
+    ngOnDestroy() {
+        this.terminator.next();
+    }
+
+    ngOnChanges(changes: SimpleChanges) {
+        if (!!changes.checkboxes) {
+            const values: ControlModel[] = changes.checkboxes.currentValue || [];
+            const currentlyFiltered = (this._checkboxes || []).filter(ctl => ctl.isFiltered).map(ctl => ctl.id);
+            const checkboxes = values.map(checkbox => new ControlModel({
+                ...checkbox,
+                isFiltered: currentlyFiltered.includes(checkbox.id),
+                label: this.translate.transform(checkbox.label || ''),
+            }));
+            this._checkboxes = this._shouldSort ? sortCheckboxes(checkboxes) : checkboxes;
+            this.isAllSelected = this._checkboxes.every(checkbox => checkbox.isSelected);
+            this.totalCount = this._checkboxes.length;
+            this.applyFilter();
+        }
+    }
+
+    toggleSelection() {
+        this.updateSelectionCount();
+        this.emitSelectionChanged();
+    }
+
+    toggleSelectAll() {
+        this.isAllSelected = !this.isAllSelected;
+        this._checkboxes = this._checkboxes.map(checkbox => new ControlModel({
+            ...checkbox,
+            isSelected: checkbox.isDisabled || (this.isFiltered && !checkbox.isFiltered) ? checkbox.isSelected : this.isAllSelected,
+        }));
+        this.applyFilter();
+        this.emitSelectionChanged();
+    }
+
+    private updateSelectionCount() {
+        this.totalSelected = this._checkboxes.filter(control => control.isSelected).length;
+        const label = 'pastanaga.' + (this.isAllSelected ? 'deselect' : 'select') + '-' + (this.isFiltered ? 'filtered' : 'all');
```

Comment:
why don't you use '`' here? I feel like it would be more readable:
```typescript
const label = `pastanaga.${this.isAllSelected ? 'deselect' : 'select'}-${this.isFiltered ? 'filtered' : 'all'}`;
```
It's a matter of taste obviously, feel free to keep your version if you prefer it

Reply:
fixed